src/Microsoft.AspNetCore.Hosting/WebHostBuilder.cs

```diff
@@ -153,7 +153,7 @@ public IWebHost Build()
 
             var hostingServices = BuildCommonServices(out var hostingStartupErrors);
             var applicationServices = hostingServices.Clone();
-            var hostingServiceProvider = hostingServices.BuildServiceProvider();
+            var hostingServiceProvider = GetProviderFromFactory(hostingServices) ?? hostingServices.BuildServiceProvider();
 
```

Comment:
At present configured factories are ignored and builder uses hardcoded ``hostingServices.BuildServiceProvider()``. Instead it should check if proper factory is registered and if yes, use it to create provider.
Fall back on hardcoded call if nothing is registered.

Reply:
I almost like this change. Lets instead build the service provider another time:

```C#
var hostingServiceProvider = hostingServices.BuildServiceProvider();
hostingServiceProvider = GetProviderFromFactory(hostingServiceProvider) ?? hostingServiceProvider;

IServiceProvider GetProviderFromFactory(IServiceProvider provider)
{
    var factory = provider.GetService<IServiceProviderFactory<IServiceCollection>>();
    return factory?.CreateServiceProvider(factory.CreateBuilder(colection));
}
```

It's a bit more robust.

Reply:
Done